File: backend/src/models/Flow.ts
import {
    Table,
    Column,
    CreatedAt,
    UpdatedAt,
    Model,
    DataType,
    PrimaryKey,
    AutoIncrement,
    ForeignKey,
    BelongsTo
} from "sequelize-typescript";
import Tenant from "./Tenant";
import User from "./User";
import Whatsapp from "./Whatsapp";

@Table
class Flow extends Model<Flow> {
    @PrimaryKey
    @AutoIncrement
    @Column
    id: number;

    @Column
    name: string;

    @Column({
        type: DataType.JSONB,
        defaultValue: []
    })
    nodes: object[];

    @Column({
        type: DataType.JSONB,
        defaultValue: []
    })
    edges: object[];

    @ForeignKey(() => Tenant)
    @Column
    tenantId: string;

    @BelongsTo(() => Tenant)
    tenant: Tenant;

    @ForeignKey(() => User)
    @Column
    @Column
    userId: number;

    @BelongsTo(() => User)
    user: User;

    @ForeignKey(() => Whatsapp)
    @Column
    whatsappId: number;

    @BelongsTo(() => Whatsapp)
    whatsapp: Whatsapp;

    @Column({ defaultValue: true })
    isActive: boolean;

    @CreatedAt
    createdAt: Date;

    @UpdatedAt
    updatedAt: Date;
}

export default Flow;

File: bussines/internal/models/flow.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
)

type Flow struct {
	ID         int            `gorm:"primaryKey" json:"id"`
	Name       string         `json:"name"`
	Nodes      datatypes.JSON `gorm:"type:json" json:"nodes"`
	Edges      datatypes.JSON `gorm:"type:json" json:"edges"`
	Active     bool           `gorm:"default:true" json:"active"`
	WhatsAppID *int           `gorm:"column:whatsappId" json:"whatsappId"`
	TenantID   uuid.UUID      `gorm:"column:tenantId;type:uuid" json:"tenantId"`
	CreatedAt  time.Time      `gorm:"column:createdAt" json:"createdAt"`
	UpdatedAt  time.Time      `gorm:"column:updatedAt" json:"updatedAt"`
}

func (Flow) TableName() string {
	return "Flows"
}
